File: Backend/Api/Services/TeamsService.cs
using Api.Models;
using Api.Repositories;

namespace Api.Services
{
    public class TeamsService:ITeamsService
    {
        private readonly  ITeamsRepository _repository;
        public TeamsService(ITeamsRepository repository)
        {
            _repository = repository;
        }
        public bool UpsertTeams(TeamsModel teams, List<ApiResponseMessages> validationMessages)
        {
            return  _repository.UpsertTeams(teams, validationMessages);
        }
        public List<TeamsModel> GetTeams(List<ApiResponseMessages> validationMessages)
        {
            return _repository.GetTeams(validationMessages);
        }
        public bool UpsertTeamMembers(TeamsModel teams,List<ApiResponseMessages> validationmessages)
        {
            return _repository.UpsertTeamMembers(teams, validationmessages);
        }
        public List<TeamsModel> GetTeamMembers(TeamsModel teams, List<ApiResponseMessages> validationMessages)
        {
            return _repository.GetTeamMembers(teams, validationMessages);
        }
    }
}

File: Frontend/src/app/core/teams.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";

@Injectable({
    providedIn:'root'
})
export class TeamsService{
    constructor(private http:HttpClient){

    }
    baseUrl = 'https://localhost:7037/'
    upsertTeams(teams){
        return this.http.post(`${this.baseUrl}api/teams/UpsertTeams`,teams);
    }
    getTeams(){
        return this.http.get(`${this.baseUrl}api/teams/GetTeams`);
    }
    upsertTeammebers(team){
        return this.http.post(`${this.baseUrl}api/teams/UpsertTeamMembers`,team);
    }
    getTeammebers(team){
        return this.http.post(`${this.baseUrl}api/teams/GetTeamMembers`,team);
    }
}